java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/ConsecutiveLiteralAppends.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2023-2025 Benoit Lacelle - SOLVEN
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.solven.cleanthat.engine.java.refactorer.mutators;
+
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.google.common.collect.ImmutableSet;
+
+import eu.solven.cleanthat.engine.java.IJdkVersionConstants;
+import eu.solven.cleanthat.engine.java.refactorer.AJavaparserExprMutator;
+import eu.solven.cleanthat.engine.java.refactorer.NodeAndSymbolSolver;
+import eu.solven.cleanthat.engine.java.refactorer.helpers.MethodCallExprHelpers;
+
+/**
+ * Switch builder.append("string").append("builder") to builder.append("stringbuilder")
+ *
+ * @author Balazs Glatz
+ */
+public class ConsecutiveLiteralAppends extends AJavaparserExprMutator {
+
+	private static final int LARGEST_SINGLE_DIGIT_NUMBER = 9;
+	private static final int SMALLEST_SINGLE_DIGIT_NUMBER = 0;
+
+	private static final String METHOD_APPEND = "append";
+
+	@Override
+	public String minimalJavaVersion() {
+		return IJdkVersionConstants.JDK_5;
+	}
+
+	@Override
+	public Set<String> getTags() {
+		return ImmutableSet.of("String");
+	}
+
+	@Override
+	public Optional<String> getPmdId() {
+		return Optional.of("ConsecutiveLiteralAppends");
+	}
+
+	@Override
+	public String pmdUrl() {
+		return "https://pmd.github.io/pmd/pmd_rules_java_performance.html#consecutiveliteralappends";
+	}
+
+	@Override
+	protected boolean processExpression(NodeAndSymbolSolver<Expression> expression) {
+		Expression node = expression.getNode();
+		if (!(node instanceof MethodCallExpr)) {
+			return false;
+		}
+
+		var methodCall = node.asMethodCallExpr();
+		if (!isAppendWithSingleParam(methodCall)) {
+			return false;
+		}
+
+		String argument = getStringValue(methodCall.getArgument(SMALLEST_SINGLE_DIGIT_NUMBER));
```

Comment:
I believe we really lack documentation to read through the main method (e.g. `processExpression`).

Typically, here, I have to:
1. Scroll up to see what's `SMALLEST_SINGLE_DIGIT_NUMBER`
2. Interprets we're parsing something like `.append(12)`
....

WAIT NO.

`SMALLEST_SINGLE_DIGIT_NUMBER` is irrelevant here, it is just some unexpected refactoring of `0` into `SMALLEST_SINGLE_DIGIT_NUMBER`.

This is my whole point about documentation: having to interpret the code due to lack of comments, I end interpreting an irrelevant code, while it is correct code. Having a snippet like `extract 'someString' from '.append("someString")'` would have helped.